### ISSUE TITLE
configure product fails with {"errors":{"base":["undefined method `identifier' for nil:NilClass"]}}

### ISSUE THREAD

**Issue by phebous** [pivotal-cf/om#304]
I have several pipelines now failing.  I have isolated all the failures happening in an radio control with incorrect yaml.  In the RabbitMQ tile when I execute an om staged-config on the staged tile, it will product this as the properties.yml config.  

```
product-name: p-rabbitmq
product-properties:
  .properties.disk_alarm_threshold:
    value: mem_relative_1_5
```

With this, I try to om --configure-product --config properties.yml which results in {"errors":{"base":["undefined method `identifier' for nil:NilClass"]}}.   Since this a required field, there is no way around this error with the exception of manually filling out this radio button and saving it.   This enables upgrades to work fine, but new installs fails.  

Here is the html control: 

\<div class="control-group" data-toggle-disabled-fields=""\>
\<div class="control-group required reverse-hereditary-focus selector"\>
\<label\>Disk free alarm limit\</label\>
\<div class="hint"\>Select the disk free limit after which RabbitMQ will raise an alarm. This value is related to the choosen vm type and persistent disk type of the RabbitMQ nodes, too low disk size or too high alarm can prevent RabbitMQ to start. Refer to the PCF RabbitMQ documentation for more details. https://docs.pivotal.io/rabbitmq-cf/1-8/install-config.html#-disk-size-concerns\</div\>
\</div\>
\<div class="accordion-group selector-options"\>
\<label\>
\<input type="radio" name="rabbitmq-broker[.properties.disk_alarm_threshold][value]" id="option_fixed50_9a62bc75-95b2-4080-ad39-b7755b4ca33e" value="fixed_50_mb" class="accordion-group-radio"\>
50MB (legacy)
\</labe\>
<\label\>
\<input type="radio" name="rabbitmq-broker[.properties.disk_alarm_threshold][value]" id="option_relative100_3acf1d8f-1b87-4381-9edd-c88995ba6138" value="mem_relative_1_0" class="accordion-group-radio"\>
100% Memory (default)
\</label\>
\<label\>
\<input type="radio" name="rabbitmq-broker[.properties.disk_alarm_threshold][value]" id="option_relative150_6d544164-bb65-4eeb-b6e2-04c2b95de74a" value="mem_relative_1_5" class="accordion-group-radio"\>
150% Memory (recommended)
\</label\>
\<label\>
\<input type="radio" name="rabbitmq-broker[.properties.disk_alarm_threshold][value]" id="option_relative200_f1b3d3de-55da-46f2-8ebf-c4a874ec9948" value="mem_relative_2_0" class="accordion-group-radio"\>
200% Memory (conservative)
\</label\>
\</div\>
\</div\>

Is this staged-config producing the wrong yaml?  What is the proper yaml to address radio input? 

om-linux 0.47
PAS 2.2.10

**Comment by phebous** [comment]
Another interesting discovery is that when referencing a radio control in the network properties, this does work.   It is just work with standard declaration.  

network-properties:
....
  singleton_availability_zone:
    name: us-west-2c

but configure-products still no dice.

**Comment by phebous** [comment]
Digging deeper, I am seeing that the the opsman api call of 

GET /api/v0/staged/products/:product_guid/properties

is failing when there isn't a anything selected from the radio list.

**Comment by jtarchie** [comment]
@phebous it has not been well documented what value should be used for selector values.

When you configure tile via the UI and then hit the API endpoint properties, you'll see that the `value` returned for the property is `150% Memory (recommended)`. For selectors, the property `value` is actually the _human readable_ version of the selected option.

At the moment, if you'd like to assign it via the config file, please use the value seen in the UI. So from above would be `200% Memory (conservative)`, `150% Memory (recommended)`, `50MB (legacy)`, or `100% Memory (default)`.

We are working with the ops manager development team to enhance the properties to be able to use the values that you were trying to assign.

Thank for documenting your issue here. This is very helpful for diagnosing user experience with the API.

**Comment by phebous** [comment]
Thanks for your response.  I am more convinced that the problem is with the opsman api vs om.  Opsman is producing a 500 error when making the GET /api/v0/staged/products/:product_guid/properties call and the deployment has a radio controll without a default value set.  OM is just passing this along.    I have also validated that once a default value is saved to deployment, it will function property.   

I have opened a case with Pivotal to update their API to address this case.  Hopefully, they will address it.
